### PR TITLE
Remove Snyk postinstall script - Closes #1568

### LINKED ISSUE #1568: Npm install fails if snyk is not installed globally
### Expected behavior

`npm install` should work without any modifications to package.json or preinstalling packages that are not listed in the Readme.md

### Actual behavior
`npm install` fails because it cannot find the `snyk` executable.
This also breaks `lisk-docker`
 
### Steps to reproduce
Use the `lisk-docker` Makefiles to build the `local` version.
`make local`

### PR DESCRIPTION
**Pull request by diego-G**
### How did I fix it?

We decide to scan our dependencies on `prepush` and `prepublishOnly`. 

### Review checklist

* The PR solves #1568
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Comment by willclarktech** [comment]
According to that issue `prepublish` is the appropriate target. A rare case where it makes more sense than `prepublishOnly`!

**Comment by willclarktech** [comment]
@karmacoma Discussed it with Diego an we thought it might make sense `prepush` instead. Otherwise, we have CI to tell us when there's a problem so `postinstall` is probably unnecessary.

File: package.json [hunk 1]
```diff
@@ -35,10 +35,9 @@
 		"docs:build": "jsdoc -c docs/conf.json --verbose --pedantic",
 		"docs:serve": "./node_modules/.bin/http-server docs/jsdoc/",
 		"precommit": "lint-staged",
-		"prepush": "npm run lint",
 		"check:dependencies": "./node_modules/.bin/snyk protect",
-		"prepublishOnly": "npm run check:dependencies",
-		"postinstall": "npm run check:dependencies"
+		"prepush": "npm run check:dependencies && npm run lint",
```
Comment (willclarktech) [inline package.json]: Prefer linting first.

Comment (diego-G) [reply package.json]: I agree.

**Pull request merged**